src/vss_tools/model.py

```diff
@@ -277,6 +281,31 @@ def check_datatype_matching_allowed_unit_datatypes(self) -> Self:
             ), f"'{self.datatype}' is not allowed for unit '{self.unit}'"
         return self
 
+    @model_validator(mode="after")
+    def check_datatype_pattern(self) -> Self:
+        """
+        Checks that regular expression datatype 'pattern' field is:
+
+        1. string only
+        2. defined only for string typed nodes i.e., STRING and STRING_ARRAY
+        3. if default value is provided it is matching the specified pattern
+        """
+        if self.pattern:
+            assert bool(type(self.pattern) is str), f"Specified pattern: '{self.pattern}' must be a string"
+
+            # Datatypes.TUPLE[0] is the string name of the type.
+            allowed_for = f"It is allowed only for '{Datatypes.STRING[0]}' and '{Datatypes.STRING_ARRAY[0]}' types."
+            assert bool(
+                Datatypes.get_type(self.datatype) in [Datatypes.STRING, Datatypes.STRING_ARRAY]
+            ), f"Field 'pattern' is not allowed for '{self.fqn}' of type: '{self.datatype}'. {allowed_for}"
+
+            if self.default:
```

Comment:
if default is a list we need to iterate and do `re.match` on each element.

It can be something like:

```python
            check_values = [self.default]
            if is_array(self.default):
                check_values = self.default
            for value in check_values:
                assert re.match(self.pattern, value), f"...."
```

Reply:
Thank you for this catch.

Reply:
Done.

Since we are going to use patterns for string arrays, probably would never happen, but just in case I also added validation of **allowed** data.